pkg/telemetry/housekeeper.go

```diff
@@ -0,0 +1,148 @@
+	}
+
+	t.stop = make(chan struct{})
+	go t.housekeeping()
+
+	return nil
+}
+
+func (t *housekeeper) Stop() error {
+	t.stop <- struct{}{}
+	close(t.stop)
+	_, err := t.lock.Unlock()
+	if err != nil {
+		return fmt.Errorf("unlock: %w", err)
+	}
+	t.lock.Close()
+	return nil
+}
+
+func (t *housekeeper) Sync() error {
+	informationTableStats, err := getInstanceInformationStats(t.engineCopy.conn)
+	if err != nil {
+		return fmt.Errorf("get instance stats: %w", err)
+	}
+	t.engineCopy.writeToTimescaleMetadataTable(convertStatsToMetadata(informationTableStats))
+	return nil
+}
+
+const housekeepingDuration = time.Hour
+
+func (t *housekeeper) housekeeping() {
+	for {
+		select {
+		case <-t.stop:
+			return
+		case <-time.After(housekeepingDuration):
+		}
+		informationTableStats, err := getInstanceInformationStats(t.engineCopy.conn)
+		if err == nil {
+			t.engineCopy.writeToTimescaleMetadataTable(convertStatsToMetadata(informationTableStats))
+		} else {
+			log.Debug("msg", "error getting instance information table stats", "error", err.Error())
+		}
+
+		if err = CleanStalePromscalesAfterCounterReset(t.engineCopy.conn); err != nil {
+			log.Debug("msg", "error cleaning stale Promscale rows", "error", err.Error())
+		}
+	}
+}
+
+type Stats map[string]int64
+
+func convertStatsToMetadata(s Stats) Metadata {
+	m := make(Metadata, len(s))
+	for k, v := range s {
+		m[trim(k, "promscale_")] = strconv.FormatInt(v, 10) // trim 'promscale_' since that will be added by the extension.
+	}
+	return m
+}
+
+func trim(s, trimStr string) string {
+	return strings.TrimPrefix(s, trimStr)
+}
+
+func getInstanceInformationStats(conn pgxconn.PgxConn) (Stats, error) {
+	stats := make(Stats)
+	var (
+		samples        int64
+		queriesExec    int64
+		queriesTimeout int64
+		queriesFailed  int64
+		traceQueryReqs int64
+		traceDepReqs   int64
+	)
+	if err := conn.QueryRow(context.Background(), `
+	SELECT
+		sum(promscale_ingested_samples_total),
+		sum(promscale_metrics_queries_executed_total),
+		sum(promscale_metrics_queries_timedout_total),
+		sum(promscale_metrics_queries_failed_total),
+		sum(promscale_trace_query_requests_executed_total),
+		sum(promscale_trace_dependency_requests_executed_total)
+	FROM _ps_catalog.promscale_instance_information`).Scan(
+		&samples, &queriesExec, &queriesTimeout, &queriesFailed, &traceQueryReqs, &traceDepReqs,
+	); err != nil {
+		return nil, fmt.Errorf("querying values from information table: %w", err)
+	}
+	stats["promscale_ingested_samples_total"] = samples
+	stats["promscale_metrics_queries_executed_total"] = queriesExec
+	stats["promscale_metrics_queries_timedout_total"] = queriesTimeout
+	stats["promscale_metrics_queries_failed_total"] = queriesFailed
+	stats["promscale_trace_query_requests_executed_total"] = traceQueryReqs
+	stats["promscale_trace_dependency_requests_executed_total"] = traceDepReqs
+	return stats, nil
+}
+
+func CleanStalePromscalesAfterCounterReset(conn pgxconn.PgxConn) error {
```

Comment:
Seems like this is exported only to be used in e2e tests. That's always a red flag to me, especially since its not supposed to be used outside of the housekeeper.

Reply:
Looking at the test a bit, this function is used for SQL function testing. 

My suggestion is to unexport this function and test the function by manually calling it like you do in this function.  Reduces the public API and causes less confusion that way.

Reply:
This function is called by housekeeping() which works on the direction of `time.After()`. Should I make a function on the struct `engineImpl` to call this, which is not part of interface as its used only for tests?

Reply:
Since the function is just running a SQL function, you can make this one private and just test the SQL function directly. That should be good enough.